### PR TITLE
Sequence on documents

### PR DESCRIPTION
**Pull request by salaesp**

### CODE REVIEW

File: src/main/java/com/recipapp/service/AbstractDocumentService.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.repository.MongoRepository;
+
+public abstract class AbstractDocumentService<O extends AbstractDocument, I> {
+
+    @Autowired
+    private SequenceService sequenceService;
+
+    public O save(O document) {
+        O modifiedDoc = this.preCreation(document);
```
Comment (JuanManuelAbate) [inline src/main/java/com/recipapp/service/AbstractDocumentService.java:14]: I don't understand what is the objective of preCreation method here.

File: src/main/java/com/recipapp/service/AbstractDocumentService.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        if (modifiedDoc.getId() == null) {
```
Comment (JuanManuelAbate) [inline src/main/java/com/recipapp/service/AbstractDocumentService.java:15]: Is there any case where modifiedDoc.getId() this isn't going to be null?

File: src/main/java/com/recipapp/service/AbstractDocumentService.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            modifiedDoc.setId(sequenceService.getNextSeq(getSequenceId()));
```
Comment (JuanManuelAbate) [inline src/main/java/com/recipapp/service/AbstractDocumentService.java]: Here in getSequenceId() you not use the "this" keyword, but in line 13 you used it. I like to use "this" always for me is more clear, what do you think?

pd: this is an stylish comment

File: src/main/java/com/recipapp/service/RecipeService.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import com.recipapp.repository.RecipeRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RecipeService extends AbstractDocumentService<Recipe, Long> {
+
+    @Autowired
+    private RecipeRepository repository;
```
Comment (JuanManuelAbate) [inline src/main/java/com/recipapp/service/RecipeService.java]: Can you change the name of the variable to recipeRepository?

File: src/main/java/com/recipapp/model/AbstractDocument.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package com.recipapp.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
```
Comment (JuanManuelAbate) [inline src/main/java/com/recipapp/model/AbstractDocument.java]: This is an unused import.

File: src/main/java/com/recipapp/model/Sequence.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.Data;
+import org.springframework.data.annotation.Id;
+
+@Data
+@AllArgsConstructor
+public class Sequence {
+
+    @Id
+    private Integer id;
+    private Long seq;
```
Comment (JuanManuelAbate) [inline src/main/java/com/recipapp/model/Sequence.java:13]: What do you think of changing this name to lastUsedIdInSequence or something similar?, maybe we can find a more representative name to this

File: src/main/java/com/recipapp/service/AbstractDocumentService.java (hunk 1 above)
Comment (salaesp) [reply src/main/java/com/recipapp/service/AbstractDocumentService.java:14]: I thought I would need it, but then i didn't. It seems it's going to be handy someday

File: src/main/java/com/recipapp/service/AbstractDocumentService.java (hunk 2 above)
Comment (salaesp) [reply src/main/java/com/recipapp/service/AbstractDocumentService.java:15]: when you are updating a document

File: src/main/java/com/recipapp/model/Sequence.java (hunk 6 above)
Comment (salaesp) [reply src/main/java/com/recipapp/model/Sequence.java:13]: This is only for the document, who has no idea how anyone is going to use it. The actual purpose is given by the service, his only objective in life is to save a sequence

**Review by JuanManuelAbate — APPROVED** [review]

**Pull request merged**